astore/client/commands/commands.go

```diff
@@ -64,10 +66,29 @@ func NewRoot(base *client.BaseFlags) *Root {
 	}
 
 	rc.store.Register(&kcobra.FlagSet{FlagSet: rc.PersistentFlags()}, "")
+	rc.Command.PersistentFlags().StringVarP(&rc.outputFile, "meta-file", "m", "",
+		fmt.Sprintf("Meta-data output file. Supported formats: %s", marshal.Formats()))
+
 	return rc
 }
 
 func (rc *Root) StoreClient() (*astore.Client, error) {
+	if rc.outputFile != "" {
+		// check output file type is supported
+		marshaller := marshal.ByExtension(rc.outputFile)
+		if marshaller == nil {
+			return nil, fmt.Errorf("Output file extension not supported `%s`.  Supported formats: %s",
+				rc.outputFile, marshal.Formats())
+		}
+
+		// check that the destination is writable
+		file, err := os.Create(rc.outputFile)
+		if err != nil {
+			return nil, fmt.Errorf("Problems creating output file `%s` - %s", rc.outputFile, err)
```

Comment:
use '%w' when printing an error, allows the caller to use Unwrap/Wrap which is a pretty cool feature of golang error handling.

(basically: fmt.Errorf turns an error into a different one, which is more appropriate for the API. But %w and Unwrap allow to access the underlying error - so the caller can be like 'was this a network error? yes, retry. Was this a file system error? ...`)

Reply:
done